api/client/commands.go

```diff
@@ -1829,10 +1829,11 @@ func (cli *DockerCli) CmdDiff(args ...string) error {
 
 func (cli *DockerCli) CmdLogs(args ...string) error {
 	var (
-		cmd    = cli.Subcmd("logs", "CONTAINER", "Fetch the logs of a container", true)
-		follow = cmd.Bool([]string{"f", "-follow"}, false, "Follow log output")
-		times  = cmd.Bool([]string{"t", "-timestamps"}, false, "Show timestamps")
-		tail   = cmd.String([]string{"-tail"}, "all", "Output the specified number of lines at the end of logs (defaults to all logs)")
+		cmd      = cli.Subcmd("logs", "CONTAINER", "Fetch the logs of a container", true)
+		follow   = cmd.Bool([]string{"f", "-follow"}, false, "Follow log output")
+		times    = cmd.Bool([]string{"t", "-timestamps"}, false, "Show timestamps")
+		tail     = cmd.String([]string{"-tail"}, "all", "Output the specified number of lines at the end of logs (defaults to all logs)")
+		truncate = cmd.Bool([]string{"-truncate"}, false, "Fetch and truncate logs of a container")
```

Comment:
`block writes -> read file -> truncate file -> unblock writes -> return read bytes` sounds _bad_.

that reads to me like we'll lose logs if the network to the client goes down any time after `truncate` and before the 10GB log is sent over the wire.

This may be acceptable in this case - I only know that in the industries I've worked, it would not be acceptable, and would cause the government to penalise the operator, and likely the software providor too.